common/av_log.c

```diff
@@ -55,6 +56,7 @@ static mp_static_mutex log_lock = MP_STATIC_MUTEX_INITIALIZER;
 static struct mpv_global *log_mpv_instance;
 static struct mp_log *log_root, *log_decaudio, *log_decvideo, *log_demuxer;
 static bool log_print_prefix = true;
+static bstr log_buffer;
```

Comment:
pretty sure this would need to be thread local
at which point you can't share an allocation and tracking it gets annoying
so a `char[4096]` isn't that bad actually

Reply:
It is locked with `mp_mutex_lock(&log_lock);`